### PR TITLE
Add initial javascript library

### PR DESCRIPTION
**Pull request by rrrliu**
Adds signal generation functions to JS to be used in burner wallet, metamask core and other wallet extensions.

Uses yarn pnp, but the amcl-js library was incompatible with installing, so files were copied over the and put into the `amcl-js` directory.

Remaining follow-ups for future changes:
- more rigorous test cases (add verification like we have in rust)
- remove unnecessary amcl-js files
- npm package registry
- linking with burner wallet
- erc spec + metamask core

### CODE REVIEW

File: javascript/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { Point } from "@noble/secp256k1";
```
Comment (rrrliu) [inline javascript/src/index.ts:1]: bulk of the logic rests in this file, `utils/` folder at the asme level, and the `test/` folder

**Comment by lsankar4033** [comment]
could publish a mirror of amcl-js at (i.e.) personaelabs/amcl-js to unbundle it from this repo

**Comment by lsankar4033** [comment]
otherwise this lgtm!

**Comment by rrrliu** [comment]
@lsankar4033 updated! Uses a published mirror package now at @rrrliu/amcl-js

**Comment by rrrliu** [comment]
> could publish a mirror of amcl-js at (i.e.) personaelabs/amcl-js to unbundle it from this repo. @Divide-By-0 lmk what you think

Updated! Uses a published mirror package now at [amcl-js](https://www.npmjs.com/package/amcl-js)

**Pull request merged**